sky/shell/platform/android/io/flutter/view/FlutterView.java

```diff
@@ -195,17 +200,23 @@ float getDevicePixelRatio() {
     public void destroy() {
         if (mPlatformServiceProviderBinding != null) {
             mPlatformServiceProviderBinding.unbind().close();
+            mPlatformServiceProviderProxy.close();
             mPlatformServiceProvider.unbindServices();
         }
 
         if (mViewServiceProviderBinding != null) {
             mViewServiceProviderBinding.unbind().close();
+            mViewServiceProviderProxy.close();
             mViewServiceProvider.unbindServices();
         }
 
         getHolder().removeCallback(mSurfaceCallback);
         nativeDetach(mNativePlatformView);
         mNativePlatformView = 0;
+
+        mSkyEngine.close();
+        mDartServiceProvider.close();
+        mFlutterAppMessages.close();
```

Comment:
You're saying that `mPlatformServiceProviderProxy` still thinks it has an owning reference to a handle after line 429?  That's very surprising and indicates that there's a serious problem in the Java bindings.  The handle that mPlatformServiceProviderProxy references should have been transferred as part of the setServices call.  The Java bindings should have done the work of passing ownership of the handle out of this object.